src/refract/PrintVisitor.cc

```diff
@@ -6,180 +6,179 @@
-        Visitor visitor(ps);                                             \
-                                                                         \
-        os << "" #ELEMENT "Element  {\n";                                \
-        for (iterator it = e.value.begin(); it != e.value.end(); ++it) { \
-            visitor.visit(*(*it));                                       \
-        }                                                                \
-        indentOS(indent);                                                \
-        os << "}\n";                                                     \
-    }                                                                    \
-
 namespace refract
 {
-    PrintVisitor::PrintVisitor()
-    : indent(0), os(std::cout) {}
+    PrintVisitor::PrintVisitor() : indent(0), os(std::cerr), ommitSrcMap(false)
+    {
+    }
 
-    PrintVisitor::PrintVisitor(int indent_, std::ostream& os_)
-    : indent(indent_), os(os_) {}
+    PrintVisitor::PrintVisitor(int indent_, std::ostream& os_,
+                               bool ommitSrcMap_)
+        : indent(indent_), os(os_), ommitSrcMap(ommitSrcMap_)
+    {
+    }
 
-    void PrintVisitor::indentOS(int ind)
+    std::ostream& PrintVisitor::indented()
     {
-        for (int i = 0; i < ind; i++) {
+        for (int i = 0; i < indent; ++i) {
             os << "  ";
         }
+        return os;
     }
 
     void PrintVisitor::printMeta(const IElement& e)
     {
-        if (e.meta.size() > 0) {
-            os << "meta {\n";
-            indentOS(indent);
-
-            for (IElement::MemberElementCollection::const_iterator i = e.meta.begin();
-                i != e.meta.end();
-                ++i) {
+        indented() << "- <meta>\n";
 
-                refract::Visit(*this, *(*i));
-            }
-
-            indentOS(indent);
-            os << "}\n";
+        for (const auto& m : e.meta) {
+            PrintVisitor{indent + 1, os, ommitSrcMap}(*m);
         }
     }
 
     void PrintVisitor::printAttr(const IElement& e)
     {
-        if (e.attributes.size() > 0) {
-            os << "attributes {\n";
-            indentOS(indent);
-
-            for (IElement::MemberElementCollection::const_iterator i = e.attributes.begin();
-                i != e.attributes.end();
-                ++i) {
+        indented() << "- <attr>\n";
 
-                refract::Visit(*this, *(*i));
-            }
-
-            indentOS(indent);
-            os << "}\n";
+        for (const auto& a : e.attributes) {
+            PrintVisitor{indent + 1, os, ommitSrcMap}(*a);
         }
     }
 
-
     void PrintVisitor::operator()(const IElement& e)
     {
-        indentOS(indent);
-        printMeta(e);
-        printAttr(e);
+        indented() << "+ " << e.element() << '\n';
+
+        PrintVisitor pv{indent + 1, os, ommitSrcMap};
+
+        refract::VisitBy(e, pv);
 
-        VisitBy(e, *this);
+        pv.printMeta(e);
+        pv.printAttr(e);
```

Comment:
Please print the meta and attribute after element name and before the visiting of content.